### DISCUSSION TITLE
Getting progress when running dbatools non-interactively?

### DISCUSSION THREAD

**Discussion by bradley-baker** [dataplat/dbatools#9255] [Q&A]
When run interactively dba powershell commandlets like Backup-DbaDatabase and Restore-DbaDatabase display a progress part with a percentage complete which is super helpful. 

However need to run the tools from a script non-interactively (as a scheduled job). I need to be able to write the progress to a log file so that I can periodically check on it. 

Is there some way to either:
- Make DBA tools print status instead of using a progress bar?
- Write some logic which checks on the status of a backup or restore job

Any assistance or pointers anyone could provide would be greatly appreciated.

**Comment by wsmelton** [comment]
Pipe the script to `Out-File` or I think use of transcripts in PowerShell also log that progress details. We are just using `Write-Progress` to achieve the progress UI in an interactive console. The built-in features to capture output in PowerShell should be able to capture that as well.
